From f7307c5bf699cd8692097eb8545c34d21214dd72 Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Tue, 2 Jul 2024 22:08:07 -0700
Subject: [PATCH] Add libraries to includes list

---
 tools/cmake/include/BoostRoot.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/cmake/include/BoostRoot.cmake b/tools/cmake/include/BoostRoot.cmake
index a0f68358..6273c716 100644
--- a/tools/cmake/include/BoostRoot.cmake
+++ b/tools/cmake/include/BoostRoot.cmake
@@ -24,7 +24,7 @@ set(__boost_incompatible_libraries "")
 if(CMAKE_SOURCE_DIR STREQUAL Boost_SOURCE_DIR)
 
   # --with-<library>
-  set(BOOST_INCLUDE_LIBRARIES "" CACHE STRING
+  set(BOOST_INCLUDE_LIBRARIES "container;json;range;system" CACHE STRING
     "List of libraries to build (default: all but excluded and incompatible)")
 
   # --without-<library>
-- 
2.37.0.windows.1

