From 3bd4854e8ab3796c65d09b52ae1f61dc84e7b11f Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Wed, 25 Dec 2024 14:41:44 -0800
Subject: [PATCH 3/3] Disable Win32 file API in Boost

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 71e746df4..7336c7c6a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -546,6 +546,7 @@ target_compile_definitions(torrent-rasterbar
 		BOOST_ASIO_NO_DEPRECATED
 		BOOST_SYSTEM_USE_UTF8
 		_SILENCE_CXX17_ALLOCATOR_VOID_DEPRECATION_WARNING
+		BOOST_BEAST_USE_WIN32_FILE=0
 	PRIVATE
 		TORRENT_BUILDING_LIBRARY
 		BOOST_EXCEPTION_DISABLE
-- 
2.49.0

